### ISSUE TITLE
python_pip fails on Debian 10

### ISSUE THREAD

**Issue by pierre-b** [robertdebock/ansible-role-python_pip#5]
Hi Robert,

I get this error while using the python_pip role on a fresh Debian:

**Playbook**

```yaml
---
- name: Prepare
  hosts: all
  gather_facts: no
  become: yes

  roles:
    - role: robertdebock.bootstrap
    - role: robertdebock.buildtools
    - role: robertdebock.python_pip
```

**Output**

TASK [robertdebock.python_pip : update pip] ************************************************************************
FAILED - RETRYING: update pip (3 retries left).
FAILED - RETRYING: update pip (2 retries left).
FAILED - RETRYING: update pip (1 retries left).
**fatal: [51.91.148.106]: FAILED! => {"attempts": 3, "changed": false, "msg": "Unable to find any of pip2, pip to use.  pip needs to be installed."}**

**Environment**
- Control node OS: MacOS
- Control node Ansible version: 2.9.6
- Managed node OS: Debian 10

Thanks for your help

**Comment by robertdebock** [comment]
It's indeed not tested, I'll try to add support for it.

Thanks for your input!

**Comment by robertdebock** [comment]
Local testing worked, let's wait CI results.

**Comment by pierre-b** [comment]
Thanks for this quick fix!

**Comment by robertdebock** [comment]
I added CI tests for Debian bullseye (I think it's version `10`) and they are [working](https://github.com/robertdebock/ansible-role-python_pip/runs/604465170?check_suite_focus=true)

I did do a lot of changes, but not specifically to the logic.

Just released [version 3.3.0 of this role](https://galaxy.ansible.com/robertdebock/python_pip). Can you have a look if that works for you?

Regards,

Robert de Bock.

**Comment by dovry** [comment]
I'm having the same problem on ubuntu 18.04, fresh off of ansible-galaxy

OS
```
% cat /etc/*release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.4 LTS"
```
Ansible
```bash
% ansible --version
ansible 2.9.7
python version = 2.7.17 (default, Apr 15 2020, 17:20:14) [GCC 7.5.0]
```
playbook
```bash
% cat prepare.yml
- name: Prepare
  hosts: localhost
  become: yes
  gather_facts: no

  roles:
    - role: robertdebock.bootstrap
    - role: robertdebock.buildtools
    - role: robertdebock.python_pip

% ansible-playbook prepare.yml

PLAY [Prepare] ***********************************************************************************************************************************************
TASK [robertdebock.bootstrap : wait for host] ****************************************************************************************************************skipping: [localhost]

TASK [robertdebock.bootstrap : test connection] **************************************************************************************************************[WARNING]: Reset is not implemented for this connection
ok: [localhost]

TASK [robertdebock.bootstrap : set bootstrap ansible_user] ***************************************************************************************************ok: [localhost]

TASK [robertdebock.bootstrap : gather ansible facts] *********************************************************************************************************ok: [localhost]

TASK [robertdebock.bootstrap : install bootstrap packages] ***************************************************************************************************ok: [localhost] => (item=python3)
ok: [localhost] => (item=sudo)
ok: [localhost] => (item=gnupg)
ok: [localhost] => (item=python3-apt)

TASK [robertdebock.buildtools : install buildtools] **********************************************************************************************************ok: [localhost]

TASK [robertdebock.python_pip : install python pip] **********************************************************************************************************ok: [localhost]

TASK [robertdebock.python_pip : update pip] ******************************************************************************************************************FAILED - RETRYING: update pip (3 retries left).
FAILED - RETRYING: update pip (2 retries left).
FAILED - RETRYING: update pip (1 retries left).
fatal: [localhost]: FAILED! => {"attempts": 3, "changed": false, "msg": "Unable to find any of pip2, pip to use.  pip needs to be installed."}

PLAY RECAP ***************************************************************************************************************************************************localhost                  : ok=6    changed=0    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0
```

I had to run `apt install python-pip` for the role to run, but that defeats the purpose.

**Comment by pierre-b** [comment]
Hi @robertdebock , finally took time to retry this, I still have the same error.
What I want to achieve is installing Zabbix with this playbook :

```yaml
---
- name: Prepare
  hosts: all
  gather_facts: yes
  become: yes

  roles:
    - role: robertdebock.bootstrap
    - role: robertdebock.buildtools
    # - role: robertdebock.python_pip
      # python_pip_modules: 
      #   - name: PyMySQL
    - role: robertdebock.container_docs
    - role: robertdebock.mysql
      mysql_databases:
        - name: zabbix
          encoding: utf8
          collation: utf8_bin
      mysql_users:
        - name: zabbix
          password: zabbix
          priv: "zabbix.*:ALL"
    - role: robertdebock.zabbix_repository
    - role: robertdebock.zabbix_server
```

But I got the following error:
`
TASK [robertdebock.mysql : create databases] ***********************************************************************
failed: [51.91.148.106] (item=zabbix) => {"ansible_loop_var": "item", "changed": false, "item": {"collation": "utf8_bin", "encoding": "utf8", "name": "zabbix"}, "msg": "The PyMySQL (Python 2.7 and Python 3.X) or MySQL-python (Python 2.X) module is required."}
`

So I added this role to my playbook after the buildtools:
```yml
    - role: robertdebock.python_pip
      python_pip_modules: 
        - name: PyMySQL
```

Which throws the following error:
```
TASK [robertdebock.python_pip : update pip] ************************************************************************
FAILED - RETRYING: update pip (3 retries left).
FAILED - RETRYING: update pip (2 retries left).
FAILED - RETRYING: update pip (1 retries left).
fatal: [51.91.148.106]: FAILED! => {"attempts": 3, "changed": false, "msg": "Unable to find any of pip2, pip to use.  pip needs to be installed."}
```

When I log into the Debian 10 machine pip is not available but pip3 is.

Any idea please?

**Comment by pierre-b** [comment]
Found out that adding the following var to my hosts solved the python issue, could you please add this to the docs?
```
[py3-hosts:vars]
ansible_python_interpreter=/usr/bin/python3
```

**Comment by robertdebock** [comment]
I think you are using an older version of Ansible that does not support [python discovery](https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html).

With [my debian latest 10/buster container](https://hub.docker.com/layers/robertdebock/debian/latest/images/sha256-dd7bd95c71511c6da0cd42600cbe040991e5c5d5ae4b21c21b85c96e6f765e19?context=repo) (based on [debian:latest](https://hub.docker.com/layers/debian/library/debian/latest/images/sha256-cdba1475474210418eef21357ffe7d03ab870f8b130cf9a9a523a2e89f7e9561?context=explore)) I run `image=debian molecule test` which succeeds.

This is also tested on [GitHub Actions](https://github.com/robertdebock/ansible-role-python_pip/runs/719493206?check_suite_focus=true) and on [Travis CI](https://travis-ci.com/github/robertdebock/ansible-role-python_pip/jobs/341266633) and succeeds.

TL;DR: Please update your Ansible version to at least 2.8.

**Comment by pierre-b** [comment]
unfortunately I'm using Ansible 2.9.6... ended up with a manual installation...
